xbmc/addons/AddonDatabase.cpp

```diff
@@ -639,15 +640,13 @@ bool CAddonDatabase::GetRepositoryContent(const std::string& id, VECADDONS& addo
       builder.SetName(m_pDS->fv("name").get_asString());
       builder.SetSummary(m_pDS->fv("summary").get_asString());
       builder.SetDescription(m_pDS->fv("description").get_asString());
+      builder.SetOrigin(m_pDS->fv("repoID").get_asString());
       DeserializeMetadata(m_pDS->fv("metadata").get_asString(), builder);
 
       auto addon = CAddonBuilder::Generate(builder.get(), ADDON_UNKNOWN);
       if (addon)
       {
-        if (!result.empty() && result.back()->ID() == addonId)
-          result.back() = std::move(addon);
-        else
-          result.push_back(std::move(addon));
+        result.emplace_back(addon);
```

Comment:
How come the old logic is no longer required?

Reply:
if e.g. `service.versioncheck` is present in both repos we would only push the version from `repository.sticky` into the private_map. now we're getting both, one in trusted and one in private

Reply:
So if we had 5 repos installed and all of them had `service.versioncheck`, what we are saying is we want all 5 not just the first one?

Reply:
i think that's the point were we would need 5 maps then, one per repo
or we have to add logic that if an addon exists in multiple repos, only the trusted passes the query..

Reply:
Ok, that’s a later PR so 😉 Maybe the next one!